### PR TITLE
not important

### PR DESCRIPTION
**Pull request by Will-Firgelli**

### CODE REVIEW

**Comment by Will-Firgelli** [comment]
Line 129 optimize for scalable use instead of hard coding

File: calibration-scalable-optical-actuators-sync-control/calibration-scalable-optical-actuators-sync-control.ino [hunk 1]
```diff
@@ -115,18 +115,34 @@ void setup() {
...
+        driveActuator(i, Direction, Speed);
+      }
     }
-  }while(prevCounter_1!=counter_1 or prevCounter_2!=counter_2);  
+  } while(compareCounter(prevCounter, counter))
+
+}
+
+// If both arrays have a similar element in the same index than return false
+bool compareCounter(int prevCounter, int counter){
```
Comment (johnkmeas) [inline calibration-scalable-optical-actuators-sync-control/calibration-scalable-optical-actuators-sync-control.ino]: I created this boolean function to iterate through 2 arrays and compare them along the same index.

File: calibration-scalable-optical-actuators-sync-control/calibration-scalable-optical-actuators-sync-control.ino [hunk 2]
```diff
@@ -115,18 +115,34 @@ void setup() {
...
+  bool areEqual = true;
+  for(int i = 0; i < numberOfActuators; i++){
+    if(prevCounter[i] == counter[i] ){
```
Comment (johnkmeas) [inline calibration-scalable-optical-actuators-sync-control/calibration-scalable-optical-actuators-sync-control.ino]: If a similar value is ever found within the two arrays change areEqual to false.

**Review by johnkmeas — COMMENTED** [review]
Created a function to dynamically compare 2 arrays and return false if similar values are found in both arrays at the same index.

File: calibration-scalable-optical-actuators-sync-control/calibration-scalable-optical-actuators-sync-control.ino [hunk 3]
```diff
@@ -79,69 +57,75 @@ void setup() {
...
-  } while(compareCounter(prevCounter, counter))
+  } while(compareCounter(prevCounter, counter));      //loop until all counts remain the same
 }
 
-// If both arrays have a similar element in the same index than return false
-bool compareCounter(int prevCounter, int counter){
-  bool areEqual = true;
+bool compareCounter(int prevCounter[], int counter[]){
+  //compares two arrays and returns false when every element of one array is the same as its corresponding indexed element in the other array
+  bool areUnequal = true;
```
Comment (Will-Firgelli) [inline calibration-scalable-optical-actuators-sync-control/calibration-scalable-optical-actuators-sync-control.ino:118]: renamed variable to suit application

File: calibration-scalable-optical-actuators-sync-control/calibration-scalable-optical-actuators-sync-control.ino [hunk 4]
```diff
@@ -79,69 +57,75 @@ void setup() {
...
   for(int i = 0; i < numberOfActuators; i++){
-    if(prevCounter[i] == counter[i] ){
-      areEqual = false;
+    if(prevCounter[i] == counter[i]){
+      areUnequal = false;
+    } 
+    else{                                             //if even one pair of elements are unequal the entire function returns true
```
Comment (Will-Firgelli) [inline calibration-scalable-optical-actuators-sync-control/calibration-scalable-optical-actuators-sync-control.ino:123]: additional condition to ensure all elements satisfy condition, not just one pair

**Pull request merged**